lua/chatgpt/api.lua

```diff
@@ -151,39 +146,58 @@ local splitCommandIntoTable = function(command)
   return cmd
 end
 
-local loadApiKey = function(command)
+local function loadConfigFromCommand(command, configName, optionName)
   local cmd = splitCommandIntoTable(command)
-  -- API KEY
   job
     :new({
       command = cmd[1],
       args = vim.list_slice(cmd, 2, #cmd),
       on_exit = function(j, exit_code)
         if exit_code ~= 0 then
-          logger.warn("Config 'api_key_cmd' did not return a value when executed")
+          logger.warn("Config '" .. optionName .. "' did not return a value when executed")
           return
         end
-        Api.OPENAI_API_KEY = j:result()[1]:gsub("%s+$", "")
+        Api[configName] = j:result()[1]:gsub("%s+$", "")
       end,
     })
     :start()
 end
 
-function Api.setup()
-  Api.OPENAI_API_KEY = os.getenv("OPENAI_API_KEY")
-  if not Api.OPENAI_API_KEY then
-    if Config.options.api_key_cmd ~= nil and Config.options.api_key_cmd ~= "" then
-      loadApiKey(Config.options.api_key_cmd)
+local function loadConfigFromEnv(envName, configName)
+  Api[configName] = os.getenv(envName)
+  if not Api[configName] then
+    logger.warn(envName .. " environment variable not set")
+    return
+  end
+  Api[configName] = Api[configName]:gsub("%s+$", "")
```

Comment:
`  local variable = os.getenv(envName)
  if not variable then
    logger.warn(envName .. " environment variable not set")
    return
  end
  Api[configName] = variable:gsub("%s+$", "")`

Reply:
should not log warning in this method, because configuration should get from plugin option as a fallback.